### DISCUSSION TITLE
How often are libraries' trunk synchronized?

### DISCUSSION THREAD

**Discussion by jll63** [compiler-explorer/compiler-explorer#6389] [Q&A] [answered]
For those libraries that give the option of selecting the trunk, in addition to pinned releases...how frequently is the trunk synchronized with the canonical repository?

**Comment by partouf** [comment]
It depends on which library exactly (and language and compiler), there are situations where the general rule doesn't apply, but: trunk versions of libraries are synchronized nightly (between 3 and 5 am UTC)

**Reply by jll63** [reply]
Thanks. The library in question is my `yomm2`. C++.

**Reply by partouf** [reply, marked as answer]
yomm2's headers are updated every night as explained.

The library binary `libyomm2.a` that you can link against is built every night from 5 am UTC (and it will take a little while for the cache to refresh) for popular compilers like GCC 13.2 and every compiler here https://conan.compiler-explorer.com/compiler_usage.html that has 1000 or more compilers per month.
For less popular compilers it only gets built every sunday night.

**Reply by jll63** [reply]
Thanks. Actually yomm2 is header-only, although it can also be built as a shared library. But [I installed it](https://github.com/compiler-explorer/compiler-explorer/pull/5347) as header-only.

**Reply by partouf** [reply]
Ah right, we did build something back in July 2023, possibly because of a missconfiguration
